util/create-attribution-doc.sh

```diff
@@ -29,23 +29,80 @@ final_license_file=THIRD-PARTY-LICENSES.txt
-# Appending aws-cdk License to THIRD-PARTY-LICENSES file
-curl https://raw.githubusercontent.com/aws/aws-cdk/main/LICENSE >> $final_license_file
-# Adding a header for Python version and License URL to THIRD-PARTY-LICENSES file
-echo -e "\nPython $cpy_version; https://raw.githubusercontent.com/python/cpython/$cpy_version/LICENSE " >> $final_license_file
-# Appending Python License to THIRD-PARTY-LICENSES file
-curl https://raw.githubusercontent.com/python/cpython/$cpy_version/LICENSE >> $final_license_file
+# Function to Append Licenses of different Packages to the THIRD-PARTY-LICENSES file
+append_to_final_license_file(){
+  
+  # Package Name, Package Version, License Type , URL for package, URL for License
+  # Adding a header to final License file with Package Name, Package Version, License Type , URL for package
+  echo -e "\n\n\n$1 \n$2 \n$3 \n$4" >> $final_license_file
+  # Appending License
+  curl $5 >> $final_license_file
+  # Some Packages have Dual Licenses 
+  if [ $# -gt 5 ]
+    then
+      curl $6 >> $final_license_file
+      curl $7 >> $final_license_file
+  fi
+  
+
+}
+
+
+
+
+# certifi 
+append_to_final_license_file "certifi" "2022.12.07" "Mozilla Public License 2.0 (MPL 2.0)" "https://github.com/certifi/python-certifi/tree/2022.12.07" "https://raw.githubusercontent.com/certifi/python-certifi/2022.12.07/LICENSE"
+
+# exception_group 
+append_to_final_license_file "exceptiongroup" $exception_group_version "MIT License" "https://github.com/agronholm/exceptiongroup" "https://raw.githubusercontent.com/agronholm/exceptiongroup/main/LICENSE"
+
+# idna 
+append_to_final_license_file "idna" $idna_version "BSD License" "https://github.com/kjd/idna" "https://raw.githubusercontent.com/kjd/idna/master/LICENSE.md"
+
+# typing_extentions
+append_to_final_license_file "typing_extensions" $typing_extentions_version "Python Software Foundation License" "https://github.com/python/typing_extensions" "https://raw.githubusercontent.com/python/typing_extensions/main/LICENSE"
+
+# Packaging
+append_to_final_license_file "packaging" $packaging_version "pache Software License; BSD License" "https://github.com/pypa/packaging" "https://raw.githubusercontent.com/pypa/packaging/main/LICENSE" "https://raw.githubusercontent.com/pypa/packaging/main/LICENSE.APACHE" "https://raw.githubusercontent.com/pypa/packaging/main/LICENSE.BSD"
+
+# pyinstaller-hooks-contrib
+# Adding a header for pyinstaller-hooks-contrib version and License URL 
+echo -e "\n\n\npyinstaller-hooks-contrib \n2022.15\nApache License 2.0; GNU General Public License \nhttps://github.com/pyinstaller/pyinstaller-hooks-contrib/archive/refs/tags/2022.15.tar.gz." >> $final_license_file
+# Appending pyinstaller-hooks-contrib 3 License files 
+wget -c https://github.com/pyinstaller/pyinstaller-hooks-contrib/archive/refs/tags/2022.15.tar.gz -O - | tar -xz
+
+cat pyinstaller-hooks-contrib-2022.15/LICENSE >> $final_license_file
```

Comment:
why only do this for the hooks package and not the others we are ignoring with pip-licenses?

Reply:
The distribution team provided this specific link for the tar file, which is why I need to unzip it and add to the final file.  For others its a direct link which I can use.

Reply:
I could directly use this link https://github.com/pyinstaller/pyinstaller-hooks-contrib/tree/2022.15, But I thought we had already decided on going with the approach of using the tar file and unzipping it.

Reply:
I have changed the link and directly taking from online using  https://github.com/pyinstaller/pyinstaller-hooks-contrib/tree/2022.15, so that I can reuse the same function

Reply:
What I mean is it seems like we are just putting a link for the license instead of the actual license content for these packages, whereas with the hooks package we put the license content into the file.  Is that correct?

Reply:
No we are putting the license content for all the packages into the final file.